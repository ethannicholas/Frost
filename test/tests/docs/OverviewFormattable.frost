interface Formattable {
    function format(fmt:String):String
}

class Weight : Formattable {
    def weight:Real

    init(weight:Real) {
        self.weight := weight
    }

    @override
    function format(fmt:String):String {
        match fmt {
            when "kg" {
                return "\{weight}kg"
            }
            when "lb" {
                return "\{weight * 2.2}lb"
            }
            otherwise {
                unreachable
            }
        }
    }
}

method main() {
    def f:Formattable := Weight(100)
    Console.printLine(f.format("kg"))
    Console.printLine(f.format("lb"))
}
