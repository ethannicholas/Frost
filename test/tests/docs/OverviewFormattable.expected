100kg
220lb
